gymnasium/envs/box2d/bipedal_walker.py

```diff
@@ -281,7 +288,80 @@ def _destroy(self):
         self.legs = []
         self.joints = []
 
+    def _process_terrain_metadata(self):
+        STATES = (1, 2, 3)
+        STUMP, STAIRS, PIT = STATES
+
+        # Defines if the terrain should be saved or copied
+        self._predefined_terrain = bool(self._terrain_metadata)
+        # Defines if the length of the grass between obstacles should be randomly distributed
+        self._terrain_grass_x_variation = self._terrain_metadata.get(
+            "x_variation", False
+        )
+        # Defines if the grass height should randomly vary
+        self._terrain_grass_y_variation = self._terrain_metadata.get(
+            "y_variation", False
+        )
+
+        if self._predefined_terrain:
+            states = self._terrain_metadata.get("states", [])
+
+            obstacles_length = []
+            for state_object in states:
+                state, metadata = state_object.values()
+                if state in STATES:
+                    if state == STUMP:
+                        obstacle_length = metadata  # Stump metadata is the stump size
+                    elif state == STAIRS:
+                        _, stair_width, stair_steps = metadata
+                        obstacle_length = (
+                            stair_width * stair_steps
+                        )  # Stairs total length
+                    elif state == PIT:
+                        obstacle_length = 4  # Default pit x size
+                    obstacles_length.append(obstacle_length)
+
+            # Total grass portion of the terrain
+            self.terrain_grass = (TERRAIN_LENGTH - sum(obstacles_length)) // len(
+                obstacles_length
+            )
+        else:
+            self.terrain_grass = TERRAIN_GRASS
+            self._terrain_metadata = dict(states=[])
+
+    def _generate_terrain_state(self, state: int) -> any:
+        GRASS, STUMP, STAIRS, PIT, _STATES_ = range(5)
+
+        if self._predefined_terrain:
+            if state == GRASS:
+                next_state = self._terrain_metadata["states"][0]
```

Comment:
Why for GRASS do we only get the first while for the next state we pop the first?